docs/04-api-reference/front-commerce-core/graphql.mdx

```diff
@@ -0,0 +1,435 @@
+
+const result = await promiseToMutationSuccesData(
+  axios.get("https://acme.com/api/products")
+);
+```
+
+## `promiseToMutationSuccessWithData`
+
+Convert a promise to a mutation success object with data transformer.
+
+```ts
+promiseToMutationSuccesWithData(promise, (transform = (v) => v));
+```
+
+Arguments:
+
+- `promise` (`Promise`): promise to execute
+- `transform` (`function`): transformation function
+
+Returns
+
+- `object`: mutation success
+  - `success`: `boolean`
+  - `data`: `any`
+
+_Example_
+
+```ts
+import { promiseToMutationSuccesWithData } from "@front-commerce/core/graphql";
+
+const result = await promiseToMutationSuccesWithData(
+  axios.get("https://acme.com/api/products"),
+  (data) => ({
+    id: product.id,
+  })
+);
+```
+
+## `withDefault404Result`
+
+Execute a promise and return a default result if the promise is rejected with a
+404 code.
+
+:::tip
+
+This is mainly used for axios request that throw an object with property
+`response.status`. If you want to use it with a promise that isn't an axios
+promise, your promise reject object must include a `response.status` property.
+
+:::
+
+```ts
+withDefault404Result(promise, defaultResult);
+```
+
+Arguments:
+
+- `promise` (`Promise`): promise to execute
+- `defaultResult` (`any`): default result to return if the promise is rejected
+  with a 404 code
+
+_Example_
+
+```ts
+import { withDefault404Result } from "@front-commerce/core/graphql";
+
+const result = await withDefault404Result(
+  axios.get("https://acme.com/api/products"),
+  []
+);
+```
+
+## `RestPagesWalker`
+
+Make an HTTP request and walk trough all paginated results to retrieve all items
+
+```ts
+RestPagesWalker =
+  (axiosInstance,
+  { url, defaultParams, pageParamName, pageSizeParamName, responseDataKey },
+  formatItem);
+```
+
+Arguments:
+
+- `axiosInstance` (`axios`): The axios instance to use
+- `url` (`string`): The url to request
+- `defaultParams` (`object`): The search params to use for request
+- `pageParamName` (`string`): The name of the search param that contains the
+  page
+- `pageSizeParamName` (`string`): The name of the search param that contains the
+  page length
+- `responseDataKey` (`string`): The response key that contains data
+- `formatItem` (`function`): The function to format each item
+
+_Example_
+
+```ts
+import axios from "axios";
+import { RestPagesWalker } from "@front-commerce/core/graphql";
+
+const axiosInstance = axios.create({
+  baseURL: "https://acme.com/api",
+});
+
+const walker = RestPagesWalker(
+  axiosInstance,
+  {
+    url: "/products",
+    defaultParams: {
+      color: "blue",
+      size: "small",
+    },
+    pageParamName: "currentPage",
+    pageSizeParamName: "pageSize",
+    responseDataKey: "products",
+  },
+  (product) => product.id
+);
+
+const allData = await walker.walk();
+```
+
+## `makePolymorphicGraphQLTypeLoader`
+
+Create a GraphQL Type Loader that accept multiple type
```

Comment:
```suggestion
Creates a GraphQL Type Loader that accept multiple types.
```

**issue:** I'm not sure `GraphQL Type Loader` is a widely used term... I feel like this section would need a bit more explanation as to what it really means.